event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/EventStreamPageService.java

```diff
@@ -0,0 +1,152 @@
+package uk.gov.justice.services.eventsourcing.source.api.service;
+
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.HEAD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.first;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.head;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.positionOf;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Direction;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Position;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.Page;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.PagingLinks;
+import uk.gov.justice.services.eventsourcing.source.api.feed.event.EventStreamEntry;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.UriInfo;
+
+@ApplicationScoped
+public class EventStreamPageService {
+
+    private static final int PATH_SEGMENT_EVENT_STREAMS = 0;
+
+    private static final long ZERO = 0L;
+
+    @Inject
+    private EventStreamService eventStreamService;
+
+    public Page<EventStreamEntry> pageOfEventStream(final String position,
+                                                    final Direction direction,
+                                                    final long pageSize,
+                                                    final UriInfo uriInfo
+    ) throws SQLException, MalformedURLException {
```

Comment:
I don't think this level should be throwing SQL Exceptions.. That should be contained from the Database tier/service/repo/dao thingy :)

Reply:
agreed, now all sql exceptions are inside jdbc repository class